vis/graphs.js

```diff
@@ -0,0 +1,148 @@
+
+            let project = Object.values(quartersinfo)[i];
+            project_desc = Object.values(project).filter(x => x.Description);
+            y_descriptions.push(project_desc);
+        }
+    });
+
+    let proj_type_num = Array(number_project_types).fill(0);
+
+    let proj_type_quarter = Array(number_quarters).fill().map(() =>
+        Array(number_project_types).fill(0),
+    );
+
+    let quarter_count = 0;
+    y_descriptions.map(quarter => {
+        quarter.map(desc => {
+            for (i in desc) {
+                if(desc[i].includes('tech')) {
+                    proj_type_num[0]++;
+                    proj_type_quarter[quarter_count][0]++;
+                }
+                else if (desc[i].includes('non')){
+                    proj_type_num[1]++;
+                    proj_type_quarter[quarter_count][1]++;
+                }
+                if(desc[i].includes('web')) {
+                    proj_type_num[7]++;
+                    proj_type_quarter[quarter_count][7]++;
+                }
+                if(desc[i].includes('app')) {
+                    proj_type_num[4]++;
+                    proj_type_quarter[quarter_count][4]++;
+                }
+                if(desc[i].includes('art')) {
+                    proj_type_num[3]++;
+                    proj_type_quarter[quarter_count][3]++;
+                }
+                else if(desc[i].includes('ar')) {
+                    proj_type_num[8]++;
+                    proj_type_quarter[quarter_count][8]++;
+                }
+                if(desc[i].includes('game')) {
+                    proj_type_num[5]++;
+                    proj_type_quarter[quarter_count][5]++;
+                }
+                if(desc[i].includes('writing')) {
+                    proj_type_num[2]++;
+                    proj_type_quarter[quarter_count][2]++;
+                }
+                if(desc[i].includes('iot')) {
+                    proj_type_num[6]++;
+                    proj_type_quarter[quarter_count][6]++;
+                }
+                if(desc[i].includes('vr')) {
+                    proj_type_num[9]++;
+                    proj_type_quarter[quarter_count][9]++;
+                }
+            }
+        });
+        quarter_count++;
+    });
+
+    let data_projects = [];
+    let index = 2;
+    let proj = [];
+    let project_names_only = ['Writing', 'Art', 'App', 'Game', 'IoT', 'Web', 'AR', 'VR'];
+    project_names_only.map(proj_type => {
+        if (proj_type == 'Writing' || proj_type == 'Art') {
+            proj = {
+                x: ['Tech', 'Non-tech'],
+                y: [0, proj_type_num[index]],
+                type: 'bar',
+                name: project_names_only[index-2],
+            };
+        }
+        else {
+            proj = {
+                x: ['Tech', 'Non-tech'],
+                y: [proj_type_num[index], 0],
+                type: 'bar',
+                name: project_names_only[index-2],
+            };
+        }
```

Comment:
you can replace this with a single statement instead of an if

```suggestion
        proj = {
            x: ['Tech', 'Non-tech'],
            y: proj_type === 'Writing || proj_type === 'Art' ? project_name_only.slice(0, 2) : project_name_only.slice(2),
            type: 'bar',
            name: project_name_only[index],
        }
```